### PR TITLE
Added `same-site` policy for session options

### PR DESCRIPTION
**Pull request by AH-dark**
```go
Store.Options(sessions.Options{
    HttpOnly: true,
    MaxAge:   7 * 86400,
    Path:     "/",
    SameSite: http.SameSiteNoneMode,
    Secure:   true,
})
```

### CODE REVIEW

**Comment by HFO4** [comment]
Why do we need this? Smaesite=None requires HTTPS to work, this will break some existing user's application.

**Comment by AH-dark** [comment]
`SameSite=None` can save us from deploying the front-end and back-end under a domain name.

You are right, there is such a problem. Maybe we can add a configuration to the conf file for this?

**Comment by AH-dark** [comment]
So should I put it in the CORS section or open a new section? How do you think it should be?

**Comment by HFO4** [comment]
> So should I put it in the CORS section or open a new section? How do you think it should be?

I think under CORS section is good.

**Comment by AH-dark** [comment]
I have finished adding the relevant configuration.

By the way, I just learned that the `Set-Cookie` header also has a directive called `Secure`, which should correspond to HTTPS. That is, if it is a cross-site request and the request side is an HTTPS site, it should be enable.

See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

**Comment by AH-dark** [comment]
@HFO4 Sir, you haven't given me an answer yet!

File: go.mod [hunk 1]
```diff
@@ -10,12 +10,12 @@ require (
 	github.com/fatih/color v1.9.0
 	github.com/gin-contrib/cors v1.3.0
 	github.com/gin-contrib/gzip v0.0.2-0.20200226035851-25bef2ef21e8
-	github.com/gin-contrib/sessions v0.0.1
+	github.com/gin-contrib/sessions v0.0.5
```
Comment (HFO4) [inline go.mod:13]: Why do we need to upgrade those pkgs?

**Review by HFO4 — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (AH-dark) [reply go.mod:13]: Older versions of the package do not provide the `SameSite` parameter

**Comment by AH-dark** [comment]
> There seems to be conflict in `go.mod`.

Solved.

**Comment by HFO4** [comment]
Thanks!

**Pull request merged**